controllers/application/deployment.go

```diff
@@ -44,6 +46,12 @@ func (r *ApplicationReconciler) reconcileDeployment(ctx context.Context, applica
 		}
 
 		r.SetLabelsFromApplication(ctx, &deployment, *application)
+		
+		// add an external link to argocd
+		ingresses := application.Spec.Ingresses
+		if len(ingresses) > 0 {
+			deployment.ObjectMeta.Annotations[AnnotationKeyLinkPrefix] =  ingresses[0]
```

Comment:
What do we do when there are multiple ingresses specified? 

Open question: Is `Deployment` the correct resource to place the annotation at?

Reply:
For reference: https://argo-cd.readthedocs.io/en/stable/user-guide/external-url/

> What do we do when there are multiple ingresses specified?

The annotation only supports one URL per resource, so my suggestion was to take the first URL and ignore the others. Then we can document that the first URL will be used for links in the UI.

> Open question: Is Deployment the correct resource to place the annotation at?

I assumed only deployments are supported from the example in the documentation above, but reading the text closer it doesn't specify that as a limitation. We could consider putting it on another resource.

Reply:
Let's start with one URL per application, at the `Deployment` level and iterate if we see other requirements emerge.